chroma_core/lib/service_config.py

```diff
@@ -620,7 +620,11 @@ def container_setup(self, username, password):
 
         self._setup_crypto()
 
-        self.try_shell(['python ./manage.py print-settings > /var/lib/chroma/iml-settings.conf'], shell=True)
+        # Many iml docker containers depend on the iml-settings file and its contents. However, redirecting the settings output
+        # into /var/lib/chroma/iml-settings.conf is not sufficient as the > operator is not atomic (the file will be created without content). 
+        # The mv command is atomic, thus the contents will be created in a temp file and then moved into /var/lib/chroma/iml-settings.conf.
+        self.try_shell(['python ./manage.py print-settings > /tmp/temp-settings.conf'], shell=True)
+        self.try_shell(['mv /tmp/temp-settings.conf /var/lib/chroma/iml-settings.conf'], shell=True)
```

Comment:
Do we need shell=True on this one?

Reply:
I saw that on the code factor. I'll try taking it out.

Reply:
```
setup_1              | Traceback (most recent call last):
setup_1              |   File "/usr/local/bin/chroma-config", line 11, in <module>
setup_1              |     load_entry_point('iml-manager==4.1.0.0', 'console_scripts', 'chroma-config')()
setup_1              |   File "/usr/local/lib/python2.7/site-packages/iml_manager-4.1.0.0-py2.7.egg/chroma_core/lib/service_config.py", line 952, in chroma_config
setup_1              |     errors = service_config.container_setup(username, password)
setup_1              |   File "/usr/local/lib/python2.7/site-packages/iml_manager-4.1.0.0-py2.7.egg/chroma_core/lib/service_config.py", line 626, in container_setup
setup_1              |     self.try_shell(['python ./manage.py print-settings > /tmp/temp-settings.conf'])
setup_1              |   File "/usr/local/lib/python2.7/site-packages/iml_manager-4.1.0.0-py2.7.egg/chroma_core/lib/util.py", line 169, in try_shell
setup_1              |     cmdline, mystdout, mystderr, stdin_text, shell=shell)
setup_1              |   File "/usr/local/lib/python2.7/site-packages/iml_manager-4.1.0.0-py2.7.egg/chroma_core/lib/util.py", line 183, in shell
setup_1              |     stderr=mystderr, stdin=stdin, shell=shell)
setup_1              |   File "/usr/local/lib/python2.7/subprocess.py", line 394, in __init__
setup_1              |     errread, errwrite)
setup_1              |   File "/usr/local/lib/python2.7/subprocess.py", line 1047, in _execute_child
setup_1              |     raise child_exception
setup_1              | OSError: [Errno 2] No such file or directory
```
line 626:
```
self.try_shell(['python ./manage.py print-settings > /tmp/temp-settings.conf'])
```

Reply:
Yeah, but I think the one on 627 can be done without it. even without shell:

https://docs.python.org/2/library/shutil.html#shutil.move

Reply:
I think the error is due to redirection. Looking at docs...

Reply:
That makes sense on 627. I'm looking for another way to do 626 as well.